### PR TITLE
Control-flow with empty body inspections and quick fixes

### PR DESCRIPTION
**Pull request by MohammadMD1383**

### CODE REVIEW

**Comment by MohammadMD1383** [comment]
#90

**Comment by MohammadMD1383** [comment]
I don't know how to make it better😅, could you help me please? @i582

**Comment by i582** [comment]
> I don't know how to make it better😅, could you help me please? @i582

Add tests 😄

**Comment by i582** [comment]
I'm not sure if it's worth highlighting them as unused, let's do it like in Kotlin and other languages, will it be a warning for the `if` or `else` keyword?

Снимок экрана 2022-10-11 в 13 03 40

And it seems that the inspection message is also not always correct, because `if` is not always an expression, it can be a statement. I also suggest changing the message as in Kotlin above.

**Comment by i582** [comment]
Let's change the name of the group to "Probable bugs" like in Kotlin. It's good practice to see how it's done for other languages.

**Review by i582 — CHANGES_REQUESTED** [review]

File: src/main/resources/inspectionDescriptions/VlangControlFlowWithEmptyBodiesInspection.html [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<html>
+<body>
+Reports empty if expressions in the code
```
Comment (i582) [inline src/main/resources/inspectionDescriptions/VlangControlFlowWithEmptyBodiesInspection.html:3]: Doesn't comply with the new checks

File: src/main/kotlin/org/vlang/ide/inspections/probableBugs/VlangControlFlowWithEmptyBodiesInspection.kt [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                )
+            }
+        }
+    }
+
+    class EmptyControlFlowQuickFix : LocalQuickFix {
+        override fun getName() = "Remove empty statement"
+        override fun getFamilyName() = name
+        override fun applyFix(project: Project, descriptor: ProblemDescriptor) {
+            val element = descriptor.psiElement!!
```
Comment (i582) [inline src/main/kotlin/org/vlang/ide/inspections/probableBugs/VlangControlFlowWithEmptyBodiesInspection.kt:58]: I usually don't use `!!` as this is a potential NPE, it seems better to do `?: return` here and below

File: src/main/kotlin/org/vlang/ide/inspections/probableBugs/VlangControlFlowWithEmptyBodiesInspection.kt [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                if (forBlock.statementList.isEmpty()) holder.registerProblem(
+                    o.`for`, "For has empty body",
+                    ProblemHighlightType.WARNING, EmptyControlFlowQuickFix()
+                )
+            }
+        }
+    }
+
+    class EmptyControlFlowQuickFix : LocalQuickFix {
+        override fun getName() = "Remove empty statement"
```
Comment (i582) [inline src/main/kotlin/org/vlang/ide/inspections/probableBugs/VlangControlFlowWithEmptyBodiesInspection.kt:55]: There is no need to override `getName`, it calls `getFamilyName` by default, so `override fun getFamilyName() = "Remove empty statement"` is enough

**Review by i582 — CHANGES_REQUESTED** [review]

**Comment by i582** [comment]
And tests of course 😄

File: src/main/kotlin/org/vlang/ide/inspections/probableBugs/VlangControlFlowWithEmptyBodiesInspection.kt (hunk 3 above)
Comment (MohammadMD1383) [reply src/main/kotlin/org/vlang/ide/inspections/probableBugs/VlangControlFlowWithEmptyBodiesInspection.kt:55]: Thanks for the tip! I didn't know that :)

File: src/test/resources/inspections/probableBugs/ctrlflow_with_empty_body.after.v [hunk 4]
```diff
@@ -0,0 +1,12 @@
+fn main() {
+
+
+	if true {
```
Comment (i582) [inline src/test/resources/inspections/probableBugs/ctrlflow_with_empty_body.after.v:4]: Seems to be a formatting problem.

Comment (MohammadMD1383) [reply src/test/resources/inspections/probableBugs/ctrlflow_with_empty_body.after.v:4]: should I handle it manually or someway trigger the formatter?

Comment (i582) [reply src/test/resources/inspections/probableBugs/ctrlflow_with_empty_body.after.v:4]: I think if you have a plugin installed in the IDE where you are developing, then calling the formatter will do everything for you

Comment (MohammadMD1383) [reply src/test/resources/inspections/probableBugs/ctrlflow_with_empty_body.after.v]: sorry for the mistake, I misunderstood your review 😂.

**Comment by MohammadMD1383** [comment]
Seems that the IDE is replacing 4-spaces with tabs after applying quick-fix

File: src/test/resources/inspections/probableBugs/ctrlflow_with_empty_body.v [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    if true {
+    } else <warning descr="If has empty body">if</warning> false {
+    }
+
+    if true {
+    } else if false {
+    } <warning descr="Else has empty body">else</warning> {
+    }
+
+    <warning descr="For has empty body">for</warning> el in arr {
```
Comment (i582) [inline src/test/resources/inspections/probableBugs/ctrlflow_with_empty_body.v:14]: Please add tests with a non-empty body and a body with a comment. Ideally, tests should cover all possible cases.

**Comment by i582** [comment]
And rename PR

**Review by i582 — APPROVED** [review]

**Comment by i582** [comment]
Thank you, this is a great contribution!
